starter-code/functions-and-arrays.js

```diff
@@ -129,9 +198,28 @@ console.log(howManyMatter);
 var howManyDog = howManyTimes(words, "dog");
 console.log(howManyDog);
 
+
+
+//
+//
 // Bonus Quest
 function greatestProduct (matrix) {
-
+  maxValue = 0 ;
+  yLength = matrix.length - 1;
+  xLength = matrix[0].length - 1 ;
+  for ( y = 1; y < yLength; y++){
```

Comment:
En este caso, estás guardando en las variables yLength y xLength un valor que solo utilizas 1 vez en el bucle for.

En lugar de almacenarlo, también estaría bien usar directamente en el bucle con los valores en lugar de variables.